src/Microsoft.AspNetCore.Razor.Language/TagHelperDescriptorExtensions.cs

```diff
@@ -7,16 +7,36 @@ namespace Microsoft.AspNetCore.Razor.Language
 {
     public static class TagHelperDescriptorExtensions
     {
-        public static string GetTypeName(this TagHelperDescriptor descriptor)
+        public static string GetTypeName(this TagHelperDescriptor tagHelper)
         {
-            descriptor.Metadata.TryGetValue(TagHelperMetadata.Common.TypeName, out var typeName);
+            if (tagHelper == null)
+            {
+                throw new ArgumentNullException(nameof(tagHelper));
+            }
 
+            tagHelper.Metadata.TryGetValue(TagHelperMetadata.Common.TypeName, out var typeName);
             return typeName;
         }
 
-        public static bool IsDefaultKind(this TagHelperDescriptor descriptor)
+        public static bool IsDefaultKind(this TagHelperDescriptor tagHelper)
         {
-            return string.Equals(descriptor.Kind, TagHelperConventions.DefaultKind, StringComparison.Ordinal);
+            if (tagHelper == null)
+            {
+                throw new ArgumentNullException(nameof(tagHelper));
+            }
+
+            return string.Equals(tagHelper.Kind, TagHelperConventions.DefaultKind, StringComparison.Ordinal);
+        }
+
+        public static bool KindUsesDefaultTagHelperRuntime(this TagHelperDescriptor tagHelper)
```

Comment:
It means that this tag helper uses things like `TagHelperExecutionContext`. Basically everything that uses the `ITagHelper` and related types.